### PR TITLE
NCC: Weathered Sentinels and support (Refactor "CARDNAME can attack as though it didn't have defender" to static)

### LINKED ISSUE #390: Weathered Sentinels

### PR DESCRIPTION
**Pull request by Northmoc**
closes #390 

Seems to be working well!

The keyword has been scrubbed from the engine!

### CODE REVIEW

File: forge-game/src/main/java/forge/game/staticability/StaticAbility.java [hunk 1]
```diff
@@ -331,6 +331,8 @@ public final boolean applyAbility(final String mode, final Card card, final Game
             return StaticAbilityCantAttackBlock.applyCantBlockByAbility(this, card, (Card)target);
         } else if (mode.equals("CanAttackIfHaste")) {
             return StaticAbilityCantAttackBlock.applyCanAttackHasteAbility(this, card, target);
+        } else if (mode.equals("CanAttackDefender")) {
+            return StaticAbilityCanAttackDefender.applyCanAttackAbility(this, card, target);
```
Comment (Hanmac) [inline forge-game/src/main/java/forge/game/staticability/StaticAbility.java:335]: better move that part to StaticAbilityCanAttackDefender like i did with some other abilities so i can remove this ambiguous "applyAbility" in the future

File: forge-gui/res/cardsfolder/d/dark_maze.txt [hunk 2]
```diff
@@ -3,6 +3,7 @@ ManaCost:4 U
 Types:Creature Wall
 PT:4/5
 K:Defender
-A:AB$ Pump | Cost$ 0 | Defined$ Self | KW$ HIDDEN CARDNAME can attack as though it didn't have defender. | AtEOT$ Exile | SpellDescription$ CARDNAME can attack this turn as though it didn't have defender. Exile it at the beginning of the next end step.
+A:AB$ Animate | Cost$ 0 | staticAbilities$ CanAttack | AtEOT$ Exile | SpellDescription$ CARDNAME can attack this turn as though it didn't have defender. Exile it at the beginning of the next end step.
```
Comment (Hanmac) [inline forge-gui/res/cardsfolder/d/dark_maze.txt:6]: Animate Effect is wrong there, it should be An Effect Effect for the Static

**Comment by Hanmac** [comment]
the ones that say: "X can attack if it hasn't defender until end of turn" should not use an Animate Effect, but an Effect in the Command Zone because an Animated Effect can be removed by loose abilities

File: forge-game/src/main/java/forge/game/combat/CombatUtil.java [hunk 3]
```diff
@@ -406,6 +406,7 @@ public static void checkDeclaredAttacker(final Game game, final Card c, final Co
         c.getDamageHistory().setCreatureAttackedThisCombat(true);
         c.getDamageHistory().clearNotAttackedSinceLastUpkeepOf();
         c.getController().addCreaturesAttackedThisTurn(CardUtil.getLKICopy(c));
+        c.getController().addAttackedPlayersMyTurn(combat.getDefenderPlayerByAttacker(c));
```
Comment (tool4ever) [inline forge-game/src/main/java/forge/game/combat/CombatUtil.java:409]: I don't think it should count if Defender is PW

**Review by tool4ever — COMMENTED** [review]
Huh, looks like the `Condition$ AttackedPlayerWhoAttackedYouLastTurn` of _O-Kagachi, Vengeful Kami_ was a bit wrong all this time
(feel free to turn this into a followup issue for me if you'd like)
👍

**Review by tool4ever — APPROVED** [review]

**Review by squee1968 — COMMENTED** [review]
I've seen some cards that have the parameter written as staticAbilities#, and some as StaticAbilities# which is correct? will the card work properly in either case?

**Comment by Northmoc** [comment]
> I've seen some cards that have the parameter written as staticAbilities#, and some as StaticAbilities# which is correct? will the card work properly in either case?

It depends on the context. Animate Effect uses the lowercase s... probably just need to change it to make it consistent

**Pull request merged**